zenoh/src/api/config.rs

```diff
@@ -0,0 +1,224 @@
+use std::{
+    any::Any,
+    error::Error,
+    fmt,
+    ops::{self, Deref},
+    path::Path,
+    sync::{Arc, Mutex, MutexGuard},
+};
+
+use serde::{Deserialize, Serialize};
+use zenoh_result::ZResult;
+
+/// Zenoh configuration.
+///
+/// Most options are optional as a way to keep defaults flexible. Some of the options have different
+/// default values depending on the rest of the configuration.
+///
+/// To construct a configuration, we advise that you use a configuration file (JSON, JSON5 and YAML
+/// are currently supported, please use the proper extension for your format as the deserializer
+/// will be picked according to it).
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct Config(pub(crate) zenoh_config::Config);
+
+impl Config {
+    pub fn from_env() -> ZResult<Self> {
+        Ok(Config(zenoh_config::Config::from_env()?))
+    }
+
+    pub fn from_file<P: AsRef<Path>>(path: P) -> ZResult<Self> {
+        Ok(Config(zenoh_config::Config::from_file(path)?))
+    }
+
+    pub fn insert_json5(&mut self, key: &str, value: &str) -> Result<(), InsertionError> {
+        self.0.insert_json5(key, value).map_err(InsertionError)
+    }
+
+    #[zenoh_macros::unstable]
+    pub fn get<'a>(&'a self, key: &str) -> Result<&'a dyn Any, LookupError> {
```

Comment:
Is this function really needed? I would remove it.

Reply:
It's only used as an example in the `Session::config` rustdoc:

https://github.com/eclipse-zenoh/zenoh/blob/fb0f4ead6283df9f74b79a9d5e06cc3227a34256/zenoh/src/api/session.rs#L636-L644

I don't have anything against removing it, the above is the only reason why I kept it.

Reply:
Deleted.